pkg/didcomm/common/service/service.go

```diff
@@ -122,3 +132,22 @@ func (m *DIDCommMsg) ThreadID() (string, error) {
 
 	return "", ErrThreadIDNotFound
 }
+
+// MessageHeader is header of the generic message
+type MessageHeader struct {
+	*Header
+	Purpose  string `json:"purpose"`
+	DataType string `json:"data-type"`
```

Comment:
I would remove.

Reply:
You mean moving them to different struct?

Reply:
I am unsure that `data-type` will exist, so it seems premature to add it :).